core/src/dird/ua_restore.cc

```diff
@@ -839,15 +848,63 @@ static bool get_date(UaContext* ua, char* date, int date_len)
   ua->SendMsg(
       _("The restored files will the most current backup\n"
         "BEFORE the date you specify below.\n\n"));
+
   for (;;) {
     if (!GetCmd(ua, _("Enter date as YYYY-MM-DD HH:MM:SS :"))) { return false; }
+    std::string cmp = CompensateShortDate(ua->cmd);
+    strcat(ua->cmd, cmp.c_str());
+
     if (StrToUtime(ua->cmd) != 0) { break; }
     ua->ErrorMsg(_("Improper date format.\n"));
   }
   bstrncpy(date, ua->cmd, date_len);
   return true;
 }
 
+/**
+ * Compensate missing date-time elements in shorter formats
+ */
+
+std::string CompensateShortDate(const char* cmd)
+{
+  std::stringstream ss;
+  ss << cmd;
+  std::string time, stime;
```

Comment:
```suggestion
  std::string time;
```

Reply:
Pls declare variables each line separately and just as close as where they are used.